File: 238-product-of-array-except-self/product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int left =1;
        int right = 1;
       int [] arr = new int[nums.length];
        for(int i=0;i<nums.length; i++){
            arr[i] = left;
            left *= nums[i];
        }
        for(int i=nums.length-1; i>=0; i--){
            arr[i] *= right;
            right *= nums[i];
        }
        return arr;
    }
}

File: 238-product-of-array-except-self/product-of-array-except-self.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        pre = 1
        post = 1
        res = [1] * len(nums)
        for i in range(len(nums)):
            res[i] = res[i]*pre
            # print(res)
            pre *= nums[i]
        
        for i in range(len(nums)-1,-1,-1):
            res[i] = res[i]*post
            post *= nums[i]
        return res